Constant expressions are incorrectly marked as invariant

```python
def constant_expressions():
    for i in range(10):
        _ = (0, 1, 2) # [loop-invariant]
        _ = [0, 1, 2] # [loop-invariant]
        _ = {"a": 1, "b": 2} # [loop-invariant]
```

Compiles to 
```
  2           0 LOAD_CONST               1 ((0, 1, 2))
              2 STORE_FAST               0 (_)

  3           4 LOAD_CONST               2 (0)
              6 LOAD_CONST               3 (1)
              8 LOAD_CONST               4 (2)
             10 BUILD_LIST               3
             12 STORE_FAST               0 (_)

  4          14 LOAD_CONST               3 (1)
             16 LOAD_CONST               4 (2)
             18 LOAD_CONST               5 (('a', 'b'))
             20 BUILD_CONST_KEY_MAP      2
             22 STORE_FAST               0 (_)
             24 LOAD_CONST               0 (None)
             26 RETURN_VALUE
```
- LOAD_CONST is as-fast as LOAD_FAST
- BUILD_LIST is as-fast as list.copy()
- BUILD_CONST_KEY_MAP is probably faster than dict.copy()

Another one for you:
```
from collections import Counter

def main():
    target = 'crate'
    target_s = set(target)
    target_c = Counter(target)
    for word in ('trace', 'cater'):
        c = Counter(word)
        if (
            target_s.issuperset(word) and
            all(target_c[k] >= v for k, v in c.items())
        ):
            print(f'{target} {word}')


if __name__ == '__main__':
    main()
``` 

Gets these warnings:
```
❯ pylint --load-plugins perflint ../perlint-test.py
************* Module hughbrown.perlint-test
/Users/hughbrown/perlint-test.py:1:0: C0103: Module name "perlint-test" doesn't conform to snake_case naming style (invalid-name)
/Users/hughbrown/perlint-test.py:1:0: C0114: Missing module docstring (missing-module-docstring)
/Users/hughbrown/perlint-test.py:3:0: C0116: Missing function or method docstring (missing-function-docstring)
/Users/hughbrown/perlint-test.py:8:8: C0103: Variable name "c" doesn't conform to snake_case naming style (invalid-name)
/Users/hughbrown/perlint-test.py:11:16: W8201: Consider moving this expression outside of the loop. (loop-invariant-statement)
/Users/hughbrown/perlint-test.py:11:16: W8201: Consider moving this expression outside of the loop. (loop-invariant-statement)
/Users/hughbrown/perlint-test.py:11:16: W8201: Consider moving this expression outside of the loop. (loop-invariant-statement)
```

Especially note line 11 where there is no invariant to hoist.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,7 @@
 
 * Renamed the global usage in loop to `loop-global-usage`
 * Added support for invariant f-strings
+* Bugfix: Doesn't highlight list, tuple or dict with only constant values as being invariant
 
 ## 0.7.1 (1st April 2022)
 

--- a/perflint/for_loop_checker.py
+++ b/perflint/for_loop_checker.py
@@ -193,6 +193,27 @@ class LoopInvariantChecker(BaseChecker):
         self._loop_assignments.append(set())
         self._ignore.append(node.test)
 
+    def _visit_sequence(self, node: Union[nodes.List, nodes.Tuple]) -> None:
+        if not node.elts:
+            self._ignore.append(node)
+        elif all(isinstance(e, nodes.Const) for e in node.elts):
+            self._ignore.append(node)
+
+    def visit_list(self, node: nodes.List) -> None:
+        self._visit_sequence(node)
+
+    def visit_tuple(self, node: nodes.Tuple) -> None:
+        self._visit_sequence(node)
+
+    def visit_dict(self, node: nodes.Dict) -> None:
+        if not node.items:
+            self._ignore.append(node)
+        elif all(
+            isinstance(k, nodes.Const) and isinstance(v, nodes.Const)
+            for k, v in node.items
+        ):
+            self._ignore.append(node)
+
     @checker_utils.check_messages("loop-invariant-statement")
     def leave_for(self, node: nodes.For) -> None:
         self._leave_loop(node)

--- a/tests/test_loop_invariant.py
+++ b/tests/test_loop_invariant.py
@@ -279,15 +279,32 @@ class TestUniqueReturnChecker(BaseCheckerTestCase):
         with self.assertNoMessages():
             self.walk(test_func)
 
-    # TODO : Fix constant checks
-    # def test_constants(self):
-    #     test_func = astroid.extract_node(
-    #         """
-    #     def test(): #@
-    #         for _ in fruits:
-    #             fruits = ["apple", "banana", "pear"]
-    #     """
-    #     )
+    def test_constants(self):
+        test_func = astroid.extract_node(
+            """
+        def test(): #@
+            for _ in fruits:
+                fruits = ["apple", "banana", "pear"]
+                fruits_tuple = ("apple", "banana", "pear")
+                _ = []
+                _ = [1, 2, 3]
+                _ = ()
+                _ = (1, 2, 3)
+        """
+        )
 
-    #     with self.assertNoMessages():
-    #         self.walk(test_func)
+        with self.assertNoMessages():
+            self.walk(test_func)
+
+    def test_constant_dictionaries(self):
+        test_func = astroid.extract_node(
+            """
+        def test(): #@
+            for _ in fruits:
+                _ = {}
+                fruits = {0: 1, 1: 2}
+        """
+        )
+
+        with self.assertNoMessages():
+            self.walk(test_func)

